bot/exts/moderation/stream.py

```diff
@@ -0,0 +1,149 @@
+import datetime
+import logging
+
+import discord
+from async_rediscache import RedisCache
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Emojis, Guild, Roles, STAFF_ROLES, VideoPermission
+from bot.converters import Expiry
+from bot.utils.scheduling import Scheduler
+from bot.utils.time import format_infraction_with_duration
+
+log = logging.getLogger(__name__)
+
+
+class Stream(commands.Cog):
+    """Grant and revoke streaming permissions from users."""
+
+    # Stores tasks to remove streaming permission
+    # User id : timestamp relation
+    task_cache = RedisCache()
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.scheduler = Scheduler(self.__class__.__name__)
+        self.reload_task = self.bot.loop.create_task(self._reload_tasks_from_redis())
+
+    def cog_unload(self) -> None:
+        """Cancel all scheduled tasks."""
+        self.reload_task.cancel()
+        self.reload_task.add_done_callback(lambda _: self.scheduler.cancel_all())
+
+    async def _remove_streaming_permission(self, member: discord.Member) -> None:
+        """Remove the streaming permission from the given Member."""
+        await self.task_cache.delete(member.id)
+        await member.remove_roles(discord.Object(Roles.video), reason="Streaming access revoked")
+
+    async def _reload_tasks_from_redis(self) -> None:
+        """Reload outstanding tasks from redis on startup, delete the task if the member has since left the server."""
+        await self.bot.wait_until_guild_available()
+        items = await self.task_cache.items()
+        for key, value in items:
+            member = self.bot.get_guild(Guild.id).get_member(key)
+
+            if not member:
+                try:
+                    member = await self.bot.get_guild(Guild.id).fetch_member(key)
+                except discord.errors.NotFound:
+                    log.debug(
+                        f"Member {key} left the guild before we could scheudle "
+                        "the revoking of their streaming permissions."
+                    )
+                    await self.task_cache.delete(key)
+                    continue
+                except discord.HTTPException as e:
+                    log.exception(f"Exception while trying to retrieve member {key} from discord\n{e}")
```

Comment:
This is redundant cause the entire traceback will already be included anyway.
```suggestion
                    log.exception(f"Exception while trying to retrieve member {key} from Discord.")
```